### PR TITLE
DOMA-1155 meter index page

### PR DESCRIPTION
**Pull request by nomerdvadcatpyat**
Meter log page.
изображение

### CODE REVIEW

File: apps/condo/lang/en.json [hunk 1]
```diff
@@ -568,6 +622,7 @@
   "UpdateTicket": "Update ticket",
   "UploadFile": "Upload file",
   "ValueIsTooShort": "The value is too short",
+  "NumberIsNotValid": "Please enter the correct number",
```
Comment (dkoviazin) [inline apps/condo/lang/en.json:625]: Please, synchronize translate lines

File: apps/condo/lang/ru.json [hunk 2]
```diff
@@ -117,14 +125,17 @@
...
   "RolesAndAccess": "Роли и доступы",
   "OneBillingWarning": "Внимание! Нельзя одновременно интегрироваться более чем с одним биллингом",
   "NoPermissionToSettings": "У вас нет прав для просмотра и редактирования данных настроек...",
   "NoPermissionToPage": "У вас нет прав для просмотра данной страницы",
   "NoMoreIntegrationsAllowed": "Нельзя одновременно интегрироваться с более чем одним биллингом. Свяжитесь с тех. поддержкой «{company}».",
   "CompanyName": "Домá",
   "ContactSupportToCancelIntegration": "Чтобы отключить интеграцию свяжитесь с тех. поддержкой «{company}».",
+  "ClearAllFilters": "Сбросить все фильтры",
   "ClickToSeeMore": "Нажмите, чтобы посмотреть подробнее",
+  "ClearAllFilters": "Сбросить все фильтры",
```
Comment (dkoviazin) [inline apps/condo/lang/ru.json:138]: ClearAllFilters - 2 times

**Review by dkoviazin — APPROVED** [review]

File: apps/condo/domains/common/components/icons/MeterLogIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import Icon from '@ant-design/icons'
+
+const MeterLogSVG = ({ width = 20, height = 20 }) => {
+    return (
+        <svg width={width} height={height} viewBox="0 0 22 13" xmlns="http://www.w3.org/2000/svg">
```
Comment (Dimitreee) [inline apps/condo/domains/common/components/icons/MeterLogIcon.tsx:6]: https://jakearchibald.github.io/svgomg/

File: apps/condo/domains/common/hooks/useMultipleFiltersModal.tsx [hunk 4]
```diff
@@ -59,9 +59,15 @@ function FilterComponent<T> ({
...
 }
 
+const FILTERS_POPUP_CONTAINER_ID = 'filtersPopupContainer'
+
 export const getModalFilterComponentByMeta = (filters, name, component: FilterComponentInfo): React.ReactElement => {
     const type = get(component, 'type')
-    const props = get(component, 'props')
+    const props = {
+        // It is necessary so that dropdowns do not go along with the screen when scrolling the modal window
+        getPopupContainer: () => document.getElementById(FILTERS_POPUP_CONTAINER_ID),
```
Comment (Dimitreee) [inline apps/condo/domains/common/hooks/useMultipleFiltersModal.tsx:68]: :))

File: apps/condo/domains/meter/components/MeterCard.tsx [hunk 5]
```diff
@@ -51,89 +69,113 @@ export const MeterCard = ({ meter, resource, name, lastMeterBillingMeterReading
...
+                                    <Space>
+                                        <Icon style={{ fontSize: '20px' }} />
+                                        <Typography.Text style={{ fontSize: fontSizes.content }} strong>
+                                            {resource && resource.name}
+                                        </Typography.Text>
+                                    </Space>
+                                </Col>
+                                {
+                                    meter.nextVerificationDate &&
+                                    dayjs(meter.nextVerificationDate).diff(dayjs(), 'month') < 2 ? (
```
Comment (Dimitreee) [inline apps/condo/domains/meter/components/MeterCard.tsx:94]: move to const, add readable condition name

File: apps/condo/domains/meter/components/MeterCard.tsx [hunk 6]
```diff
@@ -51,89 +69,113 @@ export const MeterCard = ({ meter, resource, name, lastMeterBillingMeterReading
...
+                                                <Divider />
+                                            ) : null
+                                        }
+                                        <Col span={24}>
+                                            <Row gutter={[20, 0]} align={'bottom'}>
+                                                <Col span={14}>
+                                                    <Form.Item
+                                                        name={[...getNameInArray(name), `value${tariffNumber}`]}
+                                                        label={
+                                                            `№ ${meter.number} ${meter.place ? `(${meter.place})` : ''}
```
Comment (Dimitreee) [inline apps/condo/domains/meter/components/MeterCard.tsx:131]: move to function, add readable name

File: apps/condo/domains/common/components/icons/MeterLogIcon.tsx (hunk 3 above)
Comment (nomerdvadcatpyat) [reply apps/condo/domains/common/components/icons/MeterLogIcon.tsx:6]: I ran all icons through svgomg

File: apps/condo/domains/meter/hooks/useMeterInfoModal.tsx [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        const columns = useMeterInfoModalTableColumns()
+
+        return (
+            <Modal
+                title={<MeterTitle address={address} meterNumber={meterNumber} />}
+                visible={isMeterInfoModalVisible}
+                centered
+                footer={null}
+                onCancel={() => setIsMeterInfoModalVisible(false)}
+                width={800}
```
Comment (Dimitreee) [inline apps/condo/domains/meter/hooks/useMeterInfoModal.tsx:49]: You sure we need use static width here?

File: apps/condo/pages/meter/index.tsx [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+                                <EmptyListView
+                                    label={EmptyListLabel}
+                                    message={EmptyListMessage}
+                                    createRoute='/ticket/create'
+                                    createLabel={CreateTicket} />
+                            ) :
+                            (
+                                <Row gutter={[0, 40]} align={'middle'} justify={'center'}>
+                                    <Col span={23}>
+                                        <FocusContainer style={{ padding: '16px' }}>
```
Comment (Dimitreee) [inline apps/condo/pages/meter/index.tsx:154]: see IFocusContainerProps ;)

File: apps/condo/pages/meter/index.tsx [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+    const { filters, sorters } = parseQuery(router.query)
+
+    const tableColumns = useTableColumns(filterMetas)
+    const searchMeterReadingsQuery = { ...filtersToWhere(filters), organization: { id: userOrganizationId } }
+
+    return (
+        <MetersPageContent
+            tableColumns={tableColumns}
+            searchMeterReadingsQuery={searchMeterReadingsQuery}
+            sortBy={sortersToSortBy(sorters) as SortMeterReadingsBy[]}
```
Comment (Dimitreee) [inline apps/condo/pages/meter/index.tsx:226]: i'm not sure type cast is correct here, seems like we need include generic types to utility sorter functions, same as  here apps/condo/domains/common/hooks/useSearch.ts

**Review by Dimitreee — APPROVED** [review]

**Comment by Dimitreee** [comment]
Brilliant!

File: apps/condo/domains/meter/hooks/useMeterInfoModal.tsx (hunk 7 above)
Comment (nomerdvadcatpyat) [reply apps/condo/domains/meter/hooks/useMeterInfoModal.tsx:49]: This is responsive width prop
изображение
изображение

**Pull request merged**